lib/etcdv3/connection_wrapper.rb

```diff
@@ -3,31 +3,40 @@ class ConnectionWrapper
 
     attr_accessor :connection, :endpoints, :user, :password, :token, :timeout
 
-    def initialize(timeout, *endpoints)
+    def initialize(timeout, *endpoints, allow_reconnect: true)
       @user, @password, @token = nil, nil, nil
       @timeout = timeout
       @endpoints = endpoints.map{|endpoint| Etcdv3::Connection.new(endpoint, @timeout) }
       @connection = @endpoints.first
+      @allow_reconnect = allow_reconnect
+    end
+
+    private def retry_or_raise(*args)
+      if @allow_reconnect
```

Comment:
I think it's reasonable to just check truthy/falsey like I'm doing. Insisting on it being strictly boolean (therefore not accepting `nil`) is imho not really ruby-like. But I defer to your judgment. But if you want to check it it's probably better to do in `initialize` instead of here.

Reply:
Yeah, I am cool with keeping it as is.  I've been writing mostly Golang the last few years and it's hard to re-adjust to this kind of thing